cmake/checks/check_02_system_features.cmake

```diff
@@ -67,6 +68,41 @@ IF(NOT m_LIBRARY MATCHES "-NOTFOUND")
 ENDIF()
 
 
+#
+# Check that we can use feenableexcept. Sets DEAL_II_HAVE_FP_EXCEPTIONS
+#
+# The test is a bit more complicated because we also check that no garbage
+# exception is thrown if we convert -std::numeric_limits<double>::max to a
+# string. This sadly happens with some compiler support libraries :-(
+
+INCLUDE (CheckCXXSourceRuns)
+
+CHECK_CXX_SOURCE_RUNS("
+  #include <fenv.h>
+  #include <limits>
+  #include <sstream>
+
+  int main()
+  {
+    feenableexcept(FE_DIVBYZERO|FE_INVALID);
+    std::ostringstream description;
+    const double lower_bound = -std::numeric_limits<double>::max();  
+
+    description << lower_bound;
+
+    return 0;
+  }
+  " 
+  _HAVE_FP_EXCEPTIONS)
+
+IF(_HAVE_FP_EXCEPTIONS)
+  MESSAGE(STATUS "Checking for Floating Point Exception macros -- Success")
+  SET(DEAL_II_HAVE_FP_EXCEPTIONS 1)
```

Comment:
In ASPECT I am defining a cached variable so the user can disable it if needed. Maybe that makes sense here too? See https://github.com/geodynamics/aspect/blob/master/CMakeLists.txt#L179